ecological/config.py

```diff
@@ -198,10 +69,48 @@ def from_metaclass_kwargs(cls, metaclass_kwargs: Dict) -> "Options":
             return cls(**options_kwargs)
         except TypeError as e:
             raise ValueError(
-                f"Invalid options for Config class: {metaclass_kwargs}."
+                f"Invalid options for Config class: {options_dict}."
             ) from e
 
 
+@dataclasses.dataclass
+class Variable:
+    variable_name: Optional[VariableName] = None
+    default: Any = _NO_DEFAULT
+    transform: Optional[TransformCallable] = None
+    source: Optional[Source] = None
+    wanted_type: Type = dataclasses.field(init=False)
+
+    def set_defaults(
+        self,
+        *,
+        variable_name: VariableName,
+        transform: TransformCallable,
+        source: Source,
+        wanted_type: Type,
+    ):
+        self.variable_name = self.variable_name or variable_name
+        self.transform = self.transform or transform
+        self.source = self.source or source
+        self.wanted_type = wanted_type
+
+    def get(self) -> VariableValue:
+        try:
+            raw_value = self.source[self.variable_name]
+        except KeyError as e:
+            if self.default is _NO_DEFAULT:
+                raise AttributeError(
+                    f"Configuration error: '{self.variable_name}' is not set."
+                ) from e
```

Comment:
```suggestion
                ) from None
```
I would not bother reporting the KeyError in addition to the more high-level AttributeError.